### DISCUSSION TITLE
Confused by the state of PactJS and PactWeb

### DISCUSSION THREAD

**Discussion by develmac** [pact-foundation/pact-js#786] [Q&A] [answered]
Dear Pacties! 

I am trying to add Pact (to be precise PactWeb and Pact Karma) to an Angular project. 
While after a while I got it working, the issue I still got is that the provider has to be configured in the karma.conf.js. What
about multiple providers? (As can be the case with microservices). 

Also is there any guide on how to exactly use PactJS in a browser? For instance with something like Karma and Jasmine?
I have seen that PactWeb is going to be phased out, so thats maybe not the way to go? 

Cheers, 

MaC

**Comment by mefellows** [comment, marked as answer]
The current state is that we don't have plans to continue to support Pact Web going forward (see also https://github.com/pact-foundation/pact-js/issues/626). I do believe there was a community member that had some thoughts on how we could continue to do so.

Personally, I don't believe the effort in maintaining the two interfaces is worth it. Any tests that are currently being run in Karma/Jasmine could easily be run in a Node based testing library (Jest, Mocha etc.). I understand this doesn't make everyone happy, but also from a maintainer point of view it increases the support burden and product surface area which ultimately means it takes us longer to add new features etc.

> While after a while I got it working, the issue I still got is that the provider has to be configured in the karma.conf.js. What
about multiple providers? (As can be the case with microservices).

You can setup multiple providers, the `pact` property takes an array: https://github.com/pact-foundation/karma-pact#installation

> Also is there any guide on how to exactly use PactJS in a browser?

Only `PactWeb` works in a browser, the main `Pact` interface does not (it needs to do things that can't be done in a browser, such as start servers and read files from the file system).

**Comment by develmac** [comment]
I can totally understand the maintenance aspect. So if I understand it correctly the long term and most feasible suggestion for Angular projects would probably be to use Jest instead of Jasmine/Karma? 
Then I would be able to use PactJS directly, right? 

Thx for the quick reply Matt! 


Cheers, 

MaC

**Reply by mefellows** [reply]
Yes, I would recommend that. Even if we do continue to support Karma, it is likely to lag behind the core implementation for a while so it wouldn't be a bad decision.
